server/app/durablejobs/jobs/ConvertAddressServiceAreaToArrayJob.java

```diff
@@ -0,0 +1,159 @@
+      int errorCount = 0;
+
+      // Filter to only include rows what have a service_area key that is a string type. Vastly
+      // improves the run time cutting out a large number of unneeded records.
+      String filter =
+          """
+jsonb_path_exists((object#>>'{}')::jsonb, '$.applicant.**.service_area ? (@.type() == "string")')
+""";
+
+      try (var query = database.find(ApplicantModel.class).where().raw(filter).findIterate()) {
+        while (query.hasNext()) {
+          try {
+            ApplicantModel applicant = query.next();
+            logger.debug("Converting service area for applicant id {}", applicant.id);
+            applicant.setApplicantData(processRow(applicant.getApplicantData()));
+            applicant.save(jobTransaction);
+          } catch (Exception e) {
+            errorCount++;
+            logger.error(e.getMessage(), e);
+          }
+        }
+      }
+
+      try (var query = database.find(ApplicationModel.class).where().raw(filter).findIterate()) {
+        while (query.hasNext()) {
+          try {
+            ApplicationModel application = query.next();
+            logger.debug("Converting service area for application id {}", application.id);
+            application.setApplicantData(processRow(application.getApplicantData()));
+            application.save(jobTransaction);
+          } catch (Exception e) {
+            errorCount++;
+            logger.error(e.getMessage(), e);
+          }
+        }
+      }
+
+      if (errorCount == 0) {
+        logger.info("Job succeeded");
+        jobTransaction.commit();
+      } else {
+        logger.error("Failed to convert service area. All changes undone. Errors: {}", errorCount);
+        jobTransaction.rollback();
+      }
+    }
+
+    logger.info("Run - End");
+  }
+
+  private ApplicantData processRow(ApplicantData applicantData) throws JsonProcessingException {
+    String serviceAreaName = Scalar.SERVICE_AREA.name().toLowerCase(Locale.ROOT);
+    JsonNode rootJsonNode = objectMapper.readTree(applicantData.asJsonString());
+
+    for (var questionJsonNode : rootJsonNode.findParents(serviceAreaName)) {
+      if (questionJsonNode.has(serviceAreaName)
+          && questionJsonNode.get(serviceAreaName).getNodeType() == JsonNodeType.STRING) {
+        ImmutableList<ServiceAreaInclusion> items =
+            Arrays.stream(questionJsonNode.get(serviceAreaName).asText().split(","))
+                .map(ConvertAddressServiceAreaToArrayJob::buildServiceAreaArrayFromString)
+                .collect(ImmutableList.toImmutableList());
+
+        JsonNode newJsonNode = objectMapper.valueToTree(items);
+
+        // Singular "service_area"
+        ((ObjectNode) questionJsonNode).remove(serviceAreaName);
+
+        // Plural "service_areas"
+        ((ObjectNode) questionJsonNode)
+            .set(Scalar.SERVICE_AREAS.name().toLowerCase(Locale.ROOT), newJsonNode);
```

Comment:
**question, non-blocking** would it be worth using Path.create() utils so any future SWEs looking at this code follow that pattern, instead of relying on knowing that paths are supposed to be lowercase?

Reply:
Yes. This was an oversight. Done.